.github/workflows/cypress-test-oidc-e2e.yml

```diff
@@ -0,0 +1,192 @@
+          yq -i ".config.dynamic.authc.openid_auth_domain.transport_enabled = true" config.yml
+          yq -i ".config.dynamic.authc.openid_auth_domain.order = 1" config.yml
+          yq -i ".config.dynamic.authc.openid_auth_domain.http_authenticator.type = \"openid\"" config.yml
+          yq -i ".config.dynamic.authc.openid_auth_domain.http_authenticator.challenge = false" config.yml
+          yq -i ".config.dynamic.authc.openid_auth_domain.http_authenticator.config.subject_key = \"preferred_username\"" config.yml
+          yq -i ".config.dynamic.authc.openid_auth_domain.http_authenticator.config.roles_key = \"roles\"" config.yml
+          yq -i ".config.dynamic.authc.openid_auth_domain.http_authenticator.config.openid_connect_url = \"http://localhost:8080/auth/realms/master/.well-known/openid-configuration\"" config.yml
+          yq -i ".config.dynamic.authc.openid_auth_domain.authentication_backend.type = \"noop\"" config.yml
+          cd ../../..
+
+      # TODO: REMOVE THIS ONCE ADMIN JAVA TOOL SUPPORT IT
+      - name: Write password to initialAdminPassword location
+        if: ${{ runner.os == 'Linux'}}
+        run:
+          echo admin >> ./opensearch-${{ env.OPENSEARCH_VERSION }}-SNAPSHOT/config/initialAdminPassword.txt
+        shell: bash
+
+      # Run any configuration scripts
+      - name: Run Setup Script for Linux
+        if: ${{ runner.os == 'Linux' }}
+        run: |
+          echo "running linux setup"
+          chmod +x ./setup.sh
+          ./setup.sh
+        shell: bash
+
+      # Run OpenSearch
+      - name: Run OpenSearch with plugin on Linux
+        if: ${{ runner.os == 'Linux'}}
+        run: |
+          /bin/bash -c "./opensearch-${{ env.OPENSEARCH_VERSION }}-SNAPSHOT/bin/opensearch &"
+        shell: bash
+
+      # Give the OpenSearch process some time to boot up before sending any requires, might need to increase the default time!
+      - name: Sleep while OpenSearch starts
+        uses: peternied/action-sleep@v1
+        with:
+          seconds: 30
+
+      # Verify that the server is operational
+      - name: Check OpenSearch Running on Linux
+        if: ${{ runner.os != 'Windows'}}
+        run: curl https://localhost:9200/_cat/plugins -u 'admin:admin' -k -v
+        shell: bash
+
+      - if: always()
+        run: cat ./opensearch-${{ env.OPENSEARCH_VERSION }}-SNAPSHOT/logs/opensearch.log
+        shell: bash
+
+      # OSD bootstrap
+      - name: Run Dashboard with Security Dashboards Plugin
+        uses: ./.github/actions/install-dashboards
+        with:
+          plugin_name: security-dashboards-plugin
+
+      # Configure the Dashboard for OIDC setup
+      - name: Configure and Run OpenSearch Dashboards with Cypress Test Cases
+        if: ${{ runner.os == 'Linux' }}
+        run: |
+          cd ./OpenSearch-Dashboards
+          echo 'server.host: "localhost"' >> ./config/opensearch_dashboards.yml
```

Comment:
Looks like there are 3 different ways (echo'ed to a file, filed on disk modified, cat'ed to a file) yml files are getting created, can we use a consistent process?

Reply:
all switched by using cat.